src/GeoExt/container/VectorLegend.js

```diff
@@ -163,10 +163,10 @@ Ext.define('GeoExt.container.VectorLegend', {
 
         if (this.layerRecord) {
             this.layer = this.layerRecord.getLayer();
-            if (this.layer.map) {
+            if (this.layer && this.layer.map) {
```

Comment:
I think getLayer() on a layer record will always a return something so the first check is unneeded here